pkg/sdk/logging/model/output/mattermost.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package output
+
+import (
+	"github.com/cisco-open/operator-tools/pkg/secret"
+
+	"github.com/kube-logging/logging-operator/pkg/sdk/logging/model/types"
+)
+
+// +name:"Mattermost"
+// +weight:"200"
+type _hugoMattermost interface{} //nolint:deadcode,unused
+
+// +docName:"Mattermost plugin for Fluentd"
+// Sends logs to Mattermost via webhooks.
+// More info at https://github.com/levigo-systems/fluent-plugin-mattermost
+//
+// ## Example output configurations
+// ```yaml
+// spec:
+//
+//	mattermost:
+//	  webhook_url: https://xxx.xx/hooks/xxxxxxxxxxxxxxx
+//	  channel_id: xxxxxxxxxxxxxxx
+//	  message_color: "#FFA500"
+//	  enable_tls: false
+//
+// ```
+type _docMattermost interface{} //nolint:deadcode,unused
+
+// +name:"Mattermost"
+// +url:"https://github.com/levigo-systems/fluent-plugin-mattermost"
+// +version:"0.2.2"
+// +description:"Sends logs to Mattermost via webhooks."
+// +status:"GA"
+type _metaMattermost interface{} //nolint:deadcode,unused
+
+// +kubebuilder:object:generate=true
+// +docName:"Output Config"
+type MattermostOutputConfig struct {
+	// webhook_url Incoming Webhook URI (Required for Incoming Webhook mode).
+	WebhookURL string `json:"webhook_url"`
```

Comment:
Shouldn't this value be extracted from a secret?

Reply:
You are right. Fixed